test/on_yubikey/test_piv.py

```diff
@@ -0,0 +1,121 @@
+import unittest
+from binascii import a2b_hex
+from ykman.driver_ccid import APDUError
+from ykman.piv import PivController
+from ykman.util import TRANSPORT
+from .util import (DestructiveYubikeyTestCase, missing_mode, open_device)
+
+
+DEFAULT_PIN = '123456'
+NON_DEFAULT_PIN = '654321'
+DEFAULT_PUK = '12345678'
+NON_DEFAULT_PUK = '87654321'
+DEFAULT_MANAGEMENT_KEY = '010203040506070801020304050607080102030405060708'
+NON_DEFAULT_MANAGEMENT_KEY = '010103040506070801020304050607080102030405060708'
+
+
+@unittest.skipIf(*missing_mode(TRANSPORT.CCID))
+class PivTestCase(DestructiveYubikeyTestCase):
+    pass
+
+
+class ManagementKey(PivTestCase):
+
+    def setUp(self):
+        self.dev = open_device(transports=TRANSPORT.CCID)
+        self.ctrl = PivController(self.dev.driver)
+        self.ctrl.reset()
+
+    def tearDown(self):
```

Comment:
Should we do a reset in tearDown as well? I noticed the device have a stored mgm key after running all the tests.

Reply:
I could add one to `tearDownClass`, which runs once after all the tests in the class. I'm not sure what value it would really add, though. I rather think that if other tests make asumptions about the state of the YubiKey it's probably better to enforce those assumptions in those tests' `setUp` or `setUpClass` instead of relying on other tests to clean up themselves - which most of the time won't be necessary anyway, and it takes nozero time to do the cleanup after all. :)

Reply:
Agreed, other tests should not make this kind of assumptions, but instead use their own setUp logic. This was more about having the device as "clean" as possible after all the tests are finished.

Reply:
Yeah, I guess that would be nice. Just one PIV reset costs about 1.5 seconds, though, so I'm a bit skeptical.